language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/IndexedExpressionNodeContext.java

```diff
@@ -55,7 +56,12 @@ public IndexedExpressionNodeContext() {
             completionItems.addAll(this.expressionCompletions(context));
         }
         this.sort(context, node, completionItems);
-        
+
         return completionItems;
     }
+
+    @Override
+    public boolean onPreValidation(BallerinaCompletionContext context, IndexedExpressionNode node) {
+        return context.getNodeAtCursor().lineRange().endLine().offset() >= context.getCursorPosition().getCharacter();
```

Comment:
can't we use the `context.getCursorPositionInTree` and the text offsets for this comparison?

Reply:
Yes we can. Will do

Reply:
Done